interoperator/controllers/schedulers/sfserviceinstanceupdater/sfserviceinstanceupdater_controller.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2018 The Service Fabrik Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sfserviceinstanceupdater
+
+import (
+	"context"
+	"crypto/md5"
+	"encoding/base64"
+	"encoding/json"
+
+	osbv1alpha1 "github.com/cloudfoundry-incubator/service-fabrik-broker/interoperator/api/osb/v1alpha1"
+	"github.com/cloudfoundry-incubator/service-fabrik-broker/interoperator/internal/config"
+	"github.com/cloudfoundry-incubator/service-fabrik-broker/interoperator/pkg/cluster/registry"
+	"github.com/go-logr/logr"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/retry"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+)
+
+// SFServiceInstanceUpdater reconciles a SFServiceInstance object
+type SFServiceInstanceUpdater struct {
+	client.Client
+	Log             logr.Logger
+	scheme          *runtime.Scheme
+	clusterRegistry registry.ClusterRegistry
+}
+
+// Reconcile schedules the SFServiceInstance to one SFCluster and sets the ClusterID in
+// SFServiceInstance.Spec.ClusterID. It chooses the destination cluster based on clusterSelector
+// template provided in the plan.
+func (r *SFServiceInstanceUpdater) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+	log := r.Log.WithValues("sfserviceinstance-updater", req.NamespacedName)
+
+	plan := &osbv1alpha1.SFPlan{}
+	err := r.Get(ctx, req.NamespacedName, plan)
+	if err != nil {
+		if apiErrors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	if plan.Spec.AutoUpdatable == false {
+		arrBytes := []byte{}
+		jsonBytes, _ := json.Marshal(plan.Spec)
+		arrBytes = append(arrBytes, jsonBytes...)
+		hash := md5.Sum(arrBytes)
+		currentSpecHash := base64.StdEncoding.EncodeToString(hash[:])
+		if plan.Status.SpecHash == "" {
+			err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
```

Comment:
Should we do the hash calculation also within the retry?

Reply:
makes sense. I'll change it.

Reply:
done. Can you please check if it looks good now ?